### PR TITLE
Feature image name

### PR DESCRIPTION
**Pull request by ar98nau**
- Added new boolean for imageName game variation
- Added some extra database initialization

### CODE REVIEW

File: src/main/java/cat/udl/eps/softarch/geolearning/config/AuthenticationConfig.java [hunk 1]
```diff
@@ -44,12 +48,40 @@ public void init(AuthenticationManagerBuilder auth) throws Exception {
...
+    ImageName imageName1 = new ImageName();
+    imageName1.setInstructions("Match name and image");
+    ImageNameQuestion imageNameQuestion3 = new ImageNameQuestion();
+    imageNameQuestion3.setImage("https://cdn.webshopapp.com/shops/94414/files/52385798/flag-of-japan.jpg");
+    imageNameQuestion3.setSolution("japan");
+    imageNameQuestion3.setImageName(imageName1);
+    List<ImageNameQuestion> questions1 = new ArrayList<>();
+    questions1.add(imageNameQuestion3);
+    imageName1.setQuestions(questions1);
+    imageNameRepository.save(imageName1);
```
Comment (bergacat1) [inline src/main/java/cat/udl/eps/softarch/geolearning/config/AuthenticationConfig.java]: Why are you creating all this? Is it for testing purposes? In that case, you should not be pushing it. Moreover, these objects will be created each time the application starts, which will end up with duplicates.

Comment (ar98nau) [reply src/main/java/cat/udl/eps/softarch/geolearning/config/AuthenticationConfig.java]: You are right, sorry. I delete it right now

File: src/main/java/cat/udl/eps/softarch/geolearning/domain/ImageName.java [hunk 2]
```diff
@@ -15,4 +19,9 @@
 
     @OneToMany(mappedBy = "imageName" ,cascade = CascadeType.ALL, orphanRemoval = true)
     private List<ImageNameQuestion> questions;
+
+    @Column
+    @NotNull
+    private Boolean isWrite = false;
```
Comment (bergacat1) [inline src/main/java/cat/udl/eps/softarch/geolearning/domain/ImageName.java:25]: I suppose this new attribute is for something you are implementing in the client side, so I will accept it when you PR the client side too.

Comment (ar98nau) [reply src/main/java/cat/udl/eps/softarch/geolearning/domain/ImageName.java:25]: Okey

Comment (ar98nau) [reply src/main/java/cat/udl/eps/softarch/geolearning/domain/ImageName.java:25]: PR created where we start using boolean isWrite
[Image write](https://github.com/UdL-EPS-SoftArch/GeoLearning/pull/11#issue-422146328)

File: src/main/java/cat/udl/eps/softarch/geolearning/domain/Game.java [hunk 3]
```diff
@@ -22,6 +23,10 @@
 	@Length(min = 10, max = 250)
 	private String instructions;
 
+	@ManyToOne
+	@JsonIdentityReference(alwaysAsId = true)
+	private ContentCreator creator;
+
```
Comment (ar98nau) [inline src/main/java/cat/udl/eps/softarch/geolearning/domain/Game.java:29]: PR opened for client side where this is used: [Image write](https://github.com/UdL-EPS-SoftArch/GeoLearning/pull/11#issue-422146328)

**Review by bergacat1 — APPROVED** [review]

**Pull request merged**